client/js/lounge.js

```diff
@@ -342,10 +343,18 @@ $(function() {
 				data.msg
 			]);
 
-		if (data.msg.self) {
+		// Unread marker is "hidden" (moved to bottom) if message comes from own
+		// user or if the channel is both active and focused
+		if (data.msg.self || (channel.hasClass("active") && document.hasFocus())) {
 			container
 				.find(".unread-marker")
 				.appendTo(container);
+
+			// Channel is marked as read on the server
+			socket.emit(
```

Comment:
Kind of a terrible fix. The server should know this on its own without clients sending `open` on every received message.